brayns/common/scene/Model.h

```diff
@@ -328,6 +331,24 @@ class Model
     /** Factory method to create an engine-specific material. */
     BRAYNS_API virtual MaterialPtr createMaterial(const size_t materialId,
                                                   const std::string& name) = 0;
+
+    /**
+     * Create a volume with the given dimensions, voxel spacing and data type
+     * where the are voxels are set via setVoxels() from any memory location.
+     */
+    BRAYNS_API virtual SharedDataVolumePtr createSharedDataVolume(
+        const Vector3ui& dimensions, const Vector3f& spacing,
+        const DataType type) const = 0;
+
+    /**
+     * Create a volume with the given dimensions, voxel spacing and data type
+     * where the voxels are copied via setBrick() into an optimized internal
+     * storage.
+     */
+    BRAYNS_API virtual BrickedVolumePtr createBrickedVolume(
+        const Vector3ui& dimensions, const Vector3f& spacing,
+        const DataType type) const = 0;
```

Comment:
SharedDataVolume and BrickedVolume look like OSPRay specific details leaking through the abstractions, or am I wrong?

Reply:
Yes and no. OSPRay is the one who provides those two volume types (in fact, they support even more). But those two types are still abstract in the sense if you have a simple raw volume (SharedData) or bricked volume format like UVF or BBIC.